lib/plugins/aws/provider/awsProvider.js

```diff
@@ -194,6 +194,13 @@ class AwsProvider {
     impl.addEnvironmentProfile(result, `AWS_${stageUpper}`);
 
     result.region = this.getRegion();
+
+    const deploymentBucketObject = this.serverless.service.provider.deploymentBucketObject;
+    if (deploymentBucketObject && deploymentBucketObject.serverSideEncryption
```

Comment:
This check is needed because s3 requires you to use signature signing version 4 for requests with `aws:kms` server side encryption. It looks like the s3 object defaults to the v3 signing method.

kms is Amazons key management service and it allows you to specify your own encryption keys for the encryption that will be done.

in contrast specifying `AES256` will use the encryption keys managed by s3.

also, you can specify your own encryption key id in the put object request with the `SSEKMSKeyId ` parameter, which is being read in from the config